src/client/client.cc

```diff
@@ -399,22 +444,48 @@ namespace Pistache::Http::Experimental
         if (!conn)
             throw std::runtime_error("Send request error");
 
-        auto fd = conn->fd();
+        Fd fd(conn->fdDirectOrFromSsl());
         if (fd == PS_FD_EMPTY)
         {
             PS_LOG_DEBUG_ARGS("Connection %p has empty fd", conn.get());
 
             conn->handleError("Could not send request");
             return;
         }
+        // fd is either the direct fd of 'conn', or, in the ssl case, the fd of
+        // the SslConnection
 
         PST_SSIZE_T totalWritten = 0;
         for (;;)
         {
-            const char* data               = buffer.data() + totalWritten;
-            const PST_SSIZE_T len          = buffer.size() - totalWritten;
-            const PST_SSIZE_T bytesWritten = PST_SOCK_SEND(GET_ACTUAL_FD(fd),
-                                                           data, len, 0);
+            const char* data         = buffer.data() + totalWritten;
+            const PST_SSIZE_T len    = buffer.size() - totalWritten;
+            PST_SSIZE_T bytesWritten = -1;
+
+#ifdef PISTACHE_USE_SSL
+            if (conn->isSsl())
+            {
+// Comment out PST_SSL_REQ_DBG to suppress log of send content
+#ifdef DEBUG
+// #define PST_SSL_REQ_DBG DEBUG // Comment in/out as desired
+#endif
```

Comment:
Editing source code to change functionality is a bit odd. I'd just remove this code; users can always pass `-DPST_SSL_REQ_DBG` if desired

Reply:
Fair. I do like having it there as a prompt. I added a comment pointing out that you can do `-DPST_SSL_REQ_DBG`